libbeat/esleg/eslegclient/connection.go

```diff
@@ -498,15 +500,30 @@ func (conn *Connection) execHTTPRequest(req *http.Request) (int, []byte, error)
 		req.Host = host
 	}
 
+	if conn.CompressionLevel != 0 {
+		req.Header.Add("Accept-Encoding", "gzip")
+	}
+
 	resp, err := conn.HTTP.Do(req)
 	if err != nil {
 		return 0, nil, err
 	}
 	defer closing(resp.Body, conn.log)
 
 	status := resp.StatusCode
+
+	// Check that the server actually sent compressed data
+	var reader io.ReadCloser
+	switch resp.Header.Get("Content-Encoding") {
+	case "gzip":
+		reader, err = gzip.NewReader(resp.Body)
```

Comment:
Is this actually necessary? Are we doing something to prevent this from transparently working, which is what https://pkg.go.dev/net/http#Transport suggests should happen:

```go
   // DisableCompression, if true, prevents the Transport from
	// requesting compression with an "Accept-Encoding: gzip"
	// request header when the Request contains no existing
	// Accept-Encoding value. If the Transport requests gzip on
	// its own and gets a gzipped response, it's transparently
	// decoded in the Response.Body. However, if the user
	// explicitly requested gzip it is not automatically
	// uncompressed.
	DisableCompression [bool](https://pkg.go.dev/builtin#bool)
```